app.js

```diff
@@ -1,54 +1,231 @@
-const { NODE_ENV = 'development', PORT = 5000 } = process.env
-const express = require('express')
-const { generate: generateId } = require('shortid')
-const helpers = require('./src/helpers')
-const app = express()
+const { NODE_ENV = "development", PORT = 5000 } = process.env;
+const express = require("express");
+const { generate: generateId } = require("shortid");
+const helpers = require("./src/helpers");
+const app = express();
 
-if (NODE_ENV === 'development') app.use(require('morgan')('dev'))
-app.use(require('body-parser').json())
+if (NODE_ENV === "development") app.use(require("morgan")("dev"));
+app.use(require("body-parser").json());
 
 const data = {
-  fruits: [],
-  vegetables: []
-}
+  fruits: [
+    {
+      id: "ad_-OwhQA",
+      name: "apple",
+      price: "2.99"
+    },
+    {
+      id: "DLU6Rjfon",
+      name: "banana",
+      price: "1.99"
+    },
+    {
+      id: "3YVN-Kn_z",
+      name: "kiwis",
+      price: "4.13"
+    },
+    {
+      id: "0BtTp4JRn",
+      name: "melon",
+      price: "3.33"
+    }
+  ],
+  vegetables: [
+    {
+      id: "xg7hs7vlV",
+      name: "acorn squash",
+      price: "1.33"
+    },
+    {
+      id: "0k8RBDjm0",
+      name: "artichoke",
+      price: "1.23"
+    },
+    {
+      id: "Vt2Y8l3IE",
+      name: "basil",
+      price: "1.93"
+    },
+    {
+      id: "-iWrxvN8G",
+      name: "cabbage",
+      price: "2.98"
+    }
+  ]
+};
 
-app.get('/vegetables', (req, res, next) => {
-  const { vegetables } = data
-  res.json(vegetables)
-})
+app.get("/vegetables", (req, res, next) => {
+  const { vegetables } = data;
 
-app.get('/vegetables/:id', (req, res, next) => {
-  const { vegetables } = data
-  const { id } = req.params
-  const vegetable = vegetables.find(veggie => veggie.id === id)
+  const name = req.query.name;
+
+  let myList = [];
+
+  if (!name) {
+    return res.json(vegetables);
+  }
+  console.log(name);
+
+  myList = vegetables.filter(veggie =>
+    veggie.name.includes(name.substring(1, req.query.name.length - 1))
```

Comment:
Interesting. I assume this was to get around the quotations. This will work but also return possibly more results than expected.